contrib/dimfeld/httptreemux/v5/httptreemux.go

```diff
@@ -0,0 +1,112 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package httptreemux provides functions to trace the dimfeld/httptreemux/v5 package (https://github.com/dimfeld/httptreemux).
+package httptreemux // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/dimfeld/httptreemux/v5"
+
+import (
+	"math"
+	"net/http"
+	"strings"
+
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+
+	"github.com/dimfeld/httptreemux/v5"
+)
+
+// Router is a traced version of httptreemux.TreeMux.
+type Router struct {
+	*httptreemux.TreeMux
+	config *routerConfig
+}
+
+// New returns a new router augmented with tracing.
+func New(opts ...RouterOption) *Router {
+	cfg := new(routerConfig)
+	defaults(cfg)
+	for _, fn := range opts {
+		fn(cfg)
+	}
+	if !math.IsNaN(cfg.analyticsRate) {
+		cfg.spanOpts = append(cfg.spanOpts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
+	}
+	cfg.spanOpts = append(cfg.spanOpts, tracer.Measured())
```

Comment:
A few more `spanOpts` to add:

```
cfg.spanOpts = append(cfg.spanOpts, tracer.Tag(ext.SpanKind, ext.SpanKindServer))
cfg.spanOpts = append(cfg.spanOpts, tracer.Tag(ext.Component, "dimfeld/httptreemux.v5"))
```
https://github.com/DataDog/dd-trace-go/tree/main/contrib#usage

Reply:
Same thing for `ContextRouter`

Reply:
- [ ] add missing `spanOpts` to `Router`
- [ ] add missing `spanOpts` to `ContextRouter`